src/main/webapp/src/app/map/map.component.ts

```diff
@@ -1,77 +1,93 @@
   templateUrl: `map.component.html`,
   styleUrls: ['./map.component.css']
 })
 
-export class MapComponent implements AfterViewInit {
-
+export class MapComponent {
+  
   @ViewChild("mapContainer", { static: false })
   mapContainer!: ElementRef;
   map!: google.maps.Map;
+  hiddenGems = this.appComponent.hiddenGems;
+  type!: string;
   location = {} as Location;
 
-  constructor(private locationService: LocationService) {
-      this.locationService.getLocation
+  constructor( private appComponent: AppComponent, private locationService: LocationService) {
+    this.locationService.getLocation
       .subscribe(location => {
         this.location = location;
-    })
+      })
   }
 
-  ngAfterViewInit(): void {
-    this.mapInitializer();
+  ngDoCheck() {
+    if (this.hiddenGems !== this.appComponent.hiddenGems) {
+      this.hiddenGems = this.appComponent.hiddenGems;
+      this.mapInitializer();
+    }
   }
-
+  
   mapInitializer(): void {
     // Coordinates fetched from the user's location to set the center of the map.
     let centerOfMapCoordinates = new google.maps.LatLng(this.location.lat, this.location.lng);
     let mapOptions: google.maps.MapOptions = {
       center: centerOfMapCoordinates,
-      zoom: 10
+      zoom: 15
     };
     this.map = new google.maps.Map(this.mapContainer.nativeElement, mapOptions);
+    var androidImg = '../../assets/images/android.png';
+    //Default Marker
+    let userMarker = new google.maps.Marker({
+      position: centerOfMapCoordinates,
+      map: this.map,
+      title: "user location",
+      icon : androidImg
+    });
+
+    //Adding Click event to default marker
+    userMarker.addListener("click", () => {
+      const infoWindow = new google.maps.InfoWindow({
+        content: "You Are Here!"
+      });
+      infoWindow.open(this.map, userMarker);
+    });
     this.loadMarkers();
   }
 
   loadMarkers(): void {
-    // Hard-coded dummy markers.
-    const markers = [
-      {
-        position: new google.maps.LatLng( -33.513059, 151.1234),
-        map: this.map,
-        title: "name: Quay restaurant, address: 123 Sydney Street, Sydney, star rating: 3"
-      },
-      {
-        position: new google.maps.LatLng(-33.47,151.865),
-        map: this.map,
-        title: "name: Food Van cafe, address: 123 Sydney Street, Sydney, star rating: 3"
-      }
-    ];
-
-    markers.forEach(markerInfo => {
-      // Creating a new marker object.
-      const markerObj = new google.maps.Marker({
-        ...markerInfo
-      });
+    var diamondIcon = '../../assets/images/gem.png';
 
-      // Creating a new info window with markers info.
-      const infoWindow = new google.maps.InfoWindow({
-        content: markerInfo.title,
+    // Declare array of markers to keep the fetched data from json string from Hidden Gems object.    let length = this.hiddenGems.length;
+    let length = this.hiddenGems.length;
+    for (let i = 0; i < length; i++) {
+      let data = this.hiddenGems[i],
```

Comment:
Nit - maybe call it "currentHiddenGem" rather than data?

Reply:
ok